test/unit/session/test_session.py

```diff
@@ -293,3 +293,35 @@ def test_get_retry_after_time(box_session, attempt_number, retry_after_header, e
         retry_time = box_session._get_retry_after_time(attempt_number, retry_after_header)  # pylint: disable=protected-access
     retry_time = round(retry_time, 4)
     assert retry_time == expected_result
+
+
+@pytest.mark.parametrize(
+    'test_proxy_url,test_proxy_auth, expected_proxy_dict',
+    [
+        ('http://example-proxy.com', {'user': 'test_user', 'password': 'test_password', },
```

Comment:
What happens if the user only provides one of the expected keys in the `PROXY_AUTH` dict?  Or if they put something unrelated in there (e.g. `{ 'foo': 'bar' }`)?